### PR TITLE
[JUJU-4312] Inject ctrl config service in httpserver worker

### PR DESCRIPTION
**Pull request by anvial**
This PR injects the controller config service in httpserver worker and fixes the unit-tests.

## Checklist

- [x] Code style: imports ordered, good names, simple structure, etc
- [x] Comments saying why design decisions were made
- [x] Go unit tests, with comments saying what you're testing

## QA steps

```sh
go test github.com/juju/juju/worker/httpserver/... -gocheck.v
```

### CODE REVIEW

**Review by SimonRichardson — APPROVED** [review]

File: worker/httpserver/manifold.go [hunk 1]
```diff
@@ -162,21 +180,32 @@ func (config ManifoldConfig) start(context dependency.Context) (_ worker.Worker,
 		}
 	}()
 
+	// Get controller config.
+	var dbGetter changestream.WatchableDBGetter
+	if err := context.Get(config.ChangeStreamName, &dbGetter); err != nil {
+		return nil, errors.Trace(err)
+	}
+
+	ctrlConfigService := config.NewControllerConfigService(dbGetter, loggo.GetLogger("juju.httpserver"))
```
Comment (hmlanigan) [inline worker/httpserver/manifold.go:189]:
```suggestion
	ctrlConfigService := config.NewControllerConfigService(dbGetter, config.Logger)
```

File: worker/httpserver/manifold.go [hunk 2]
```diff
@@ -4,6 +4,7 @@
 package httpserver
 
 import (
+	stdcontext "context"
```
Comment (hmlanigan) [inline worker/httpserver/manifold.go:7]: Please be consistent with the naming, I've seen context, ctx, and stdcontext in different workers so far.

File: worker/httpserver/manifold.go [hunk 3]
```diff
@@ -162,21 +180,32 @@ func (config ManifoldConfig) start(context dependency.Context) (_ worker.Worker,
...
 
+	// Get controller config.
+	var dbGetter changestream.WatchableDBGetter
+	if err := context.Get(config.ChangeStreamName, &dbGetter); err != nil {
+		return nil, errors.Trace(err)
+	}
+
+	ctrlConfigService := config.NewControllerConfigService(dbGetter, loggo.GetLogger("juju.httpserver"))
+
+	controllerConfig, err := ctrlConfigService.ControllerConfig(stdcontext.Background())
```
Comment (hmlanigan) [inline worker/httpserver/manifold.go:191]: have we settled on context.Background()?

File: worker/httpserver/manifold.go [hunk 4]
```diff
@@ -162,21 +180,32 @@ func (config ManifoldConfig) start(context dependency.Context) (_ worker.Worker,
...
+	if err != nil {
+		return nil, errors.Trace(err)
```
Comment (hmlanigan) [inline worker/httpserver/manifold.go:193]: Inconsistent error messages for this failure across workers.

**Pull request closed without merging**